pkg/authorization/global_authorizer.go

```diff
@@ -58,8 +58,13 @@ func NewGlobalAuthorizer(localKubeInformers, globalKubeInformers kcpkubernetesin
 }
 
 func (a *GlobalAuthorizer) RulesFor(ctx context.Context, user user.Info, namespace string) ([]authorizer.ResourceRuleInfo, []authorizer.NonResourceRuleInfo, bool, error) {
-	// TODO: wire context in RulesFor interface
-	panic("implement me")
+	cluster := genericapirequest.ClusterFrom(ctx)
+	if cluster == nil || cluster.Name.Empty() {
+		return nil, nil, false, fmt.Errorf("empty cluster name")
+	}
+
+	scopedAuth := a.newAuthorizer(cluster.Name)
+	return scopedAuth.RulesFor(ctx, user, namespace)
```

Comment:
How about a small e2e to ensure `SelfSubjectAccessReview` works?

Reply:
Sounds sensible. Will take a look at that.